src/python/pants/util/dirutil.py

```diff
@@ -125,32 +127,66 @@ def safe_file_dump(filename, payload, binary_mode=None, mode=None):
     else:
       mode = 'wb'
 
-  with safe_open(filename, mode=mode) as f:
-    f.write(payload)
+  safe_file_write(filename, payload=payload, mode=mode)
 
 
-def maybe_read_file(filename, binary_mode=True):
+def safe_file_write(filename, payload=None, mode=None):
+  """Write a string to a file.
+
+  This method is "safe" to the extent that `safe_open` is "safe". See the explanation on the method
+  doc there.
+
+  When `payload` is an empty string (the default), this method can be used as a concise way to
+  create an empty file along with its containing directory (or truncate it if it already exists).
+
+  :param string filename: The filename of the file to write to.
+  :param string payload: The string to write to the file. Defaults to the empty string.
```

Comment:
With official type hints, this would be `Union[str, bytes]`. I'm not sure if there's a way to specify this with docstring style type hints? We want to express it can be bytes, whereas `filename` and `mode` must always be a Py3 `str` (unicode).

Reply:
In https://pythonhosted.org/an_example_pypi_project/sphinx.html#full-code-example I'm not seeing anything representing a union, but I will add that as an additional note. I've also [left a comment on #6742](https://github.com/pantsbuild/pants/issues/6742#issuecomment-456210569).

Reply:
Thanks for investigating!

Not sure we need to spend much effort finding how to document Union types via docstring type hints. We'll be able to use native type hints immediately once we drop Py2—even before I setup the MyPy task and CI etc, we can start adding type hints right away.

Also we don't seem to enforce docstring type hints, and they seem inconsistent. Sometimes it's called `string`, sometime `str`, for example.

More motivation to get Py3 migration done quicker.